src/simulators/statevector/qvintrin_avx.hpp

```diff
@@ -0,0 +1,913 @@
+{\
+  real_ret = _mm256_mul_ps(real_left, real_right);\
+  imag_ret = _mm256_mul_ps(real_left, imag_right);\
+  real_ret = _mm256_fnmadd_ps(imag_left, imag_right, real_ret);\
+  imag_ret = _mm256_fmadd_ps(imag_left, real_right, imag_ret);\
+}
+
+#define _mm_complex_multiply_add_ps(real_ret, imag_ret, real_left, imag_left, real_right, imag_right)\
+{\
+  real_ret = _mm256_fmadd_ps(real_left, real_right, real_ret);\
+  imag_ret = _mm256_fmadd_ps(real_left, imag_right, imag_ret);\
+  real_ret = _mm256_fnmadd_ps(imag_left, imag_right, real_ret);\
+  imag_ret = _mm256_fmadd_ps(imag_left, real_right, imag_ret);\
+}
+
+#define _mm_complex_inner_product_ps(\
+    dim, vreals, vimags, cmplxs, vret_real, vret_imag, vtmp_0, vtmp_1)\
+{\
+  vtmp_0 = _mm256_set1_ps(cmplxs[0]);\
+  vtmp_1 = _mm256_set1_ps(cmplxs[1]);\
+  _mm_complex_multiply_ps(vret_real, vret_imag, vreals[0], vimags[0], vtmp_0, vtmp_1);\
+  for (unsigned ____i = 1; ____i < dim; ++____i) {\
+    vtmp_0 = _mm256_set1_ps(cmplxs[____i * 2]);\
+    vtmp_1 = _mm256_set1_ps(cmplxs[____i * 2 + 1]);\
+    _mm_complex_multiply_add_ps(vret_real, vret_imag, vreals[____i], vimags[____i], vtmp_0, vtmp_1);\
+  }\
+}
+
+#define _mm_load_twoarray_complex_ps(real_addr_0, imag_addr_1, real_ret, imag_ret)\
+{\
+  real_ret = _mm256_load_ps(real_addr_0);\
+  imag_ret = _mm256_load_ps(imag_addr_1);\
+}
+
+#define _mm_store_twoarray_complex_ps(real_ret, imag_ret, cmplx_addr_0, cmplx_addr_1)\
+{\
+  _mm256_store_ps(cmplx_addr_0, real_ret);\
+  _mm256_store_ps(cmplx_addr_1, imag_ret);\
+}
+
+#define _mm_altload_complex_ps(cmplx_addr_0, cmplx_addr_1, real_ret, imag_ret)\
+{\
+  real_ret = _mm256_load_ps(cmplx_addr_0);\
+  imag_ret = _mm256_load_ps(cmplx_addr_1);\
+  tmp0 = _mm256_permutevar8x32_ps(real_ret, _mm256_set_epi32(6, 7, 4, 5, 2, 3, 0, 1));\
+  tmp1 = _mm256_permutevar8x32_ps(imag_ret, _mm256_set_epi32(6, 7, 4, 5, 2, 3, 0, 1));\
+  real_ret = _mm256_blend_ps(real_ret, tmp1, 0b10101010);\
+  imag_ret = _mm256_blend_ps(tmp0, imag_ret, 0b10101010);\
+  real_ret = _mm256_permutevar8x32_ps(real_ret, _mm256_set_epi32(7, 5, 3, 1, 6, 4, 2, 0));\
+  imag_ret = _mm256_permutevar8x32_ps(imag_ret, _mm256_set_epi32(7, 5, 3, 1, 6, 4, 2, 0));\
+}
+
+#define _mm_altstore_complex_ps(real_ret, imag_ret, cmplx_addr_0, cmplx_addr_1)\
+{\
+  real_ret = _mm256_permutevar8x32_ps(real_ret, _mm256_set_epi32(7, 3, 6, 2, 5, 1, 4, 0));\
+  imag_ret = _mm256_permutevar8x32_ps(imag_ret, _mm256_set_epi32(7, 3, 6, 2, 5, 1, 4, 0));\
+  tmp0 = _mm256_permutevar8x32_ps(real_ret, _mm256_set_epi32(6, 7, 4, 5, 2, 3, 0, 1));\
+  tmp1 = _mm256_permutevar8x32_ps(imag_ret, _mm256_set_epi32(6, 7, 4, 5, 2, 3, 0, 1));\
+  real_ret = _mm256_blend_ps(real_ret, tmp1, 0b10101010);\
+  imag_ret = _mm256_blend_ps(tmp0, imag_ret, 0b10101010);\
+  _mm256_store_ps(cmplx_addr_0, real_ret);\
+  _mm256_store_ps(cmplx_addr_1, imag_ret);\
+}
+
+#define _mm_altload_complex_pd(cmplx_addr_0, cmplx_addr_1, real_ret, imag_ret)\
```

Comment:
Looks like this function and the function below: `_mm_load_twoarray_complex_pd` should be pretty similar, but they look very different. Is `_mm_load_twoarray_complex_pd` correct? 
They are being called in the same context,  but the only difference is the two first arguments.